SoundSwitch.Audio.Manager/AudioSwitcher.cs

```diff
@@ -177,6 +177,23 @@ public void SwitchForegroundProcessTo(string deviceId, ERole role, EDataFlow flo
             SwitchProcessTo(deviceId, role, flow, processId);
         }
 
+        /// <summary>
+        /// Set the same master volume level from the default audio device to the next device
+        /// </summary>
+        /// <param name="deviceId">Id of the device</param>
+        /// <param name="role">Which role to switch</param>
+        /// <param name="flow">Which flow to switch</param>
+        public void SetVolumeFromDefaultDevice (DeviceInfo device) {
+            MMDeviceEnumerator devEnum = new();
+            MMDevice defaultDevice = devEnum.GetDefaultAudioEndpoint(device.Type, (Role) ERole.eConsole);
+            var currentVolume = defaultDevice.AudioEndpointVolume.MasterVolumeLevelScalar;
+            var IsMute = defaultDevice.AudioEndpointVolume.Mute;
+            
+            var nextDevice = devEnum.GetDevice(device.Id);
+            nextDevice.AudioEndpointVolume.MasterVolumeLevelScalar = currentVolume;
+            nextDevice.AudioEndpointVolume.Mute = IsMute;
+        }
+
```

Comment:
```suggestion
        /// <summary>
        /// Get the current default endpoint
        /// </summary>
        /// <param name="flow"></param>
        /// <param name="role"></param>
        /// <returns>Null if no default device is defined</returns>
        private MMDevice? GetMmDefaultAudioEndpoint(EDataFlow flow, ERole role) => ComThread.Invoke(() => EnumeratorClient.GetDefaultEndpoint(flow, role));

        /// <summary>
        /// Set the same master volume level from the default audio device to the next device
        /// </summary>
        /// <param name="device"></param>
        public void SetVolumeFromDefaultDevice(DeviceInfo device)
        {
            var currentDefault = GetMmDefaultAudioEndpoint((EDataFlow)device.Type, ERole.eConsole);
            if (currentDefault == null)
                return;
            var audioInfo = InteractWithMmDevice(currentDefault, mmDevice =>
            {
                var defaultDeviceAudioEndpointVolume = mmDevice.AudioEndpointVolume;
                return defaultDeviceAudioEndpointVolume == null ? default : (Volume: defaultDeviceAudioEndpointVolume.MasterVolumeLevelScalar, IsMuted: defaultDeviceAudioEndpointVolume.Mute);
            });

            if (audioInfo == default)
                return;

            var nextDevice = GetDevice(device.Id);
            
            if(nextDevice == null)
                return;
            
            InteractWithMmDevice(nextDevice, mmDevice =>
            {
                if (mmDevice is not { State: DeviceState.Active })
                    return nextDevice;
                
                if (mmDevice.AudioEndpointVolume == null)
                    return nextDevice;

                mmDevice.AudioEndpointVolume.MasterVolumeLevelScalar = audioInfo.Volume;
                mmDevice.AudioEndpointVolume.Mute = audioInfo.IsMuted;
                return mmDevice;
            });
        }


```

Reply:
I've rewritten this part of the code because it needs to always run in the ComThread to avoid any issue accessing the different object.

I try to avoid having any code that touch the audio device running anywhere else because they are COM object that get attached to the thread where they are created.

I've also added proper null check because so many part of this code could actually return null depending of what device is given.